src/ripple/app/consensus/RCLConsensus.cpp

```diff
@@ -0,0 +1,891 @@
+        // ordered logically "sooner" than transactions not mentioned
+        // in the previous consensus round.
+        //
+        bool anyDisputes = false;
+        for (auto& it : disputes_)
+        {
+            if (!it.second.getOurVote ())
+            {
+                // we voted NO
+                try
+                {
+                    JLOG (j_.debug())
+                        << "Test applying disputed transaction that did"
+                        << " not get in";
+
+                    SerialIter sit (it.second.tx().tx_.slice());
+                    auto txn = std::make_shared<STTx const>(sit);
+                    retriableTxs.insert (txn);
+
+                    anyDisputes = true;
+                }
+                catch (std::exception const&)
+                {
+                    JLOG (j_.debug())
+                        << "Failed to apply transaction we voted NO on";
+                }
+            }
+        }
+
+        // Build new open ledger
+        auto lock = make_lock(
+            app_.getMasterMutex(), std::defer_lock);
+        auto sl = make_lock(
+            ledgerMaster_.peekMutex (), std::defer_lock);
+        std::lock(lock, sl);
+
+        auto const lastVal = ledgerMaster_.getValidatedLedger();
+        boost::optional<Rules> rules;
+        if (lastVal)
+            rules.emplace(*lastVal);
+        else
+            rules.emplace();
+        app_.openLedger().accept(app_, *rules,
+            sharedLCL.ledger_, localTxs_.getTxSet(), anyDisputes, retriableTxs, tapNONE,
+                "consensus",
+                    [&](OpenView& view, beast::Journal j)
+                    {
+                        // Stuff the ledger with transactions from the queue.
+                        return app_.getTxQ().accept(app_, view);
+                    });
+
+    }
+
+    //-------------------------------------------------------------------------
+    {
+        ledgerMaster_.switchLCL (sharedLCL.ledger_);
+
+        // Do these need to exist?
+        assert (ledgerMaster_.getClosedLedger()->info().hash == sharedLCL.id());
+        assert (app_.openLedger().current()->info().parentHash == sharedLCL.id());
+    }
+
+    //-------------------------------------------------------------------------
+    if (haveCorrectLCL_ && ! consensusFail_)
+    {
+        // we entered the round with the network,
+        // see how close our close time is to other node's
+        //  close time reports, and update our clock.
+        JLOG (j_.info())
+            << "We closed at " << closeTime_.time_since_epoch().count();
+        using usec64_t = std::chrono::duration<std::uint64_t>;
+        usec64_t closeTotal = std::chrono::duration_cast<usec64_t>(closeTime_.time_since_epoch());
+        int closeCount = 1;
+
+        for (auto const& p : closeTimes_)
+        {
+            // FIXME: Use median, not average
+            JLOG (j_.info())
+                << beast::lexicalCastThrow <std::string> (p.second)
```

Comment:
This was just maintaining what was there before, but I agree that throwing doesn't make sense anyway. 

However, @vinniefalco has advocated for `boost::lexical_cast` instead of `std::to_string`.  Should I go with that instead?